### PR TITLE
add validate method in vault service

### PR DESCRIPTION
**Pull request by randmonkey**
Add `Validate()` method to call `/schemas/vaults/validate` to validate vaults.

### CODE REVIEW

File: kong/vault_service.go [hunk 1]
```diff
@@ -152,3 +156,28 @@ func (s *VaultService) ListAll(ctx context.Context) ([]*Vault, error) {
 	}
 	return vaults, nil
 }
+
+func (s *VaultService) Validate(ctx context.Context, vault *Vault) (bool, string, error) {
+	endpoint := "/schemas/vaults/validate"
```
Comment (pmalek) [inline kong/vault_service.go:161]: Can we make this a `const`?

File: kong/vault_service.go [hunk 2]
```diff
@@ -152,3 +156,28 @@ func (s *VaultService) ListAll(ctx context.Context) ([]*Vault, error) {
 	}
 	return vaults, nil
 }
+
+func (s *VaultService) Validate(ctx context.Context, vault *Vault) (bool, string, error) {
```
Comment (programmer04) [inline kong/vault_service.go:160]: Could it be written like a Validate [here](https://github.com/Kong/go-kong/blob/abc671e70baee66723ed806889930d07b65f2a67/kong/route_service.go#L216C1-L230)? I think it's more consistent

Comment (pmalek) [reply kong/vault_service.go:160]: 👍 Although I'd mention what the return values are. One could deduce what the `bool` is for but string isn't obvious here IMHO.

Comment (programmer04) [reply kong/vault_service.go:160]: Yup it ought to be explained in docstring

File: kong/vault_service.go (hunk 1 above)
Comment (randmonkey) [reply kong/vault_service.go]: done

File: kong/vault_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (randmonkey) [reply kong/vault_service.go:163]: done.

Comment (randmonkey) [reply kong/vault_service.go:163]: Done.

**Review by programmer04 — APPROVED** [review]

File: kong/vault_service.go [hunk 3]
```diff
@@ -152,3 +156,23 @@ func (s *VaultService) ListAll(ctx context.Context) ([]*Vault, error) {
 	}
 	return vaults, nil
 }
+
+// Validate validates a vault against its schema.
+// returns validate result (passed/failed) and the message from the schema validation service if validation fails.
+// returns a non-nil error if failed to call the schema validation service.
```
Comment (pmalek) [inline kong/vault_service.go:162]: `returns` -> `Returns`

**Pull request merged**